plugin/trace/trace.go

```diff
@@ -86,21 +101,13 @@ func (t *trace) Name() string { return "trace" }
 
 // ServeDNS implements the plugin.Handle interface.
 func (t *trace) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
-	trace := false
-	if t.every > 0 {
-		queryNr := atomic.AddUint64(&t.count, 1)
-
-		if queryNr%t.every == 0 {
-			trace = true
-		}
-	}
 	span := ot.SpanFromContext(ctx)
-	if !trace || span != nil {
+	if span != nil {
 		return plugin.NextOrFailure(t.Name(), t.Next, ctx, w, r)
 	}
 
 	req := request.Request{W: w, Req: r}
-	span = t.Tracer().StartSpan(spanName(ctx, req))
+	span = t.Tracer().StartSpan(defaultTopLevelSpanName)
```

Comment:
So, I just realized a potential problem. I am wondering about the performance impact here. Previously, because the `every` was implemented outside, we never allocated a span unless we were hitting the sample. Now, we allocate a span for every single query. What is the cost of allocating a span, in memory and time? Can you check some benchmarks before and after?

Reply:
Will check this and get back to you with graphs.

Reply:
Doesn't look good there is a 20% hit
Memory
![image](https://user-images.githubusercontent.com/5513509/95139223-4a56d880-076c-11eb-967b-f2ef6e770d3b.png)

CPU
![image](https://user-images.githubusercontent.com/5513509/95139337-8a1dc000-076c-11eb-998f-c2ab0999a80f.png)

Corresponding req/s
![image](https://user-images.githubusercontent.com/5513509/95139377-a6216180-076c-11eb-964d-aa7cc691e0bf.png)

Will leave that up a little to see if it change

Will touch base with the integration team at datadog if that's expected.

Reply:
After letting it sit for the night, things are definitely not looking good
40% increase in memory usage with a every at 100 using the datadog provider.
Currently touching base to see if the lib has a way to decrease that significantly else i will rollback that change and go back to the purest expression of the sampling we had before 

🟠  Baseline (yesterday values)
🔵  Current values
![image](https://user-images.githubusercontent.com/5513509/95184273-6c317900-07c7-11eb-86cf-98b94ceb288c.png)

![image](https://user-images.githubusercontent.com/5513509/95184368-89fede00-07c7-11eb-9a2a-1765537d6705.png)